proxy-node-shared/src/main/java/uk/gov/ida/notification/validations/Base64XmlValidator.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.ida.notification.validations;
+
+import org.apache.commons.lang3.StringUtils;
+import org.glassfish.jersey.internal.util.Base64;
+import org.w3c.dom.Document;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+public class Base64XmlValidator implements ConstraintValidator<ValidBase64Xml, String> {
+
+    @Override
+    public void initialize(ValidBase64Xml constraint) { /* intentionally blank */ }
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        if (StringUtils.isBlank(value)) { return true; } // @NotNull should detect nulls
+
+        String xml = Base64.decodeAsString(value);
+        if (StringUtils.isEmpty(xml)) { return false; }
+
+        try {
+            InputStream stream = new ByteArrayInputStream(xml.getBytes());
+            DocumentBuilder parser = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+            Document document = parser.parse(stream);
```

Comment:
Is there a check for checking if the document is empty?

Reply:
This is checking to see if the content is an XML document. All XML documents must have a root element or they'll throw during parsing. We could add an additional validation for "and cannot be a valid XML document with only the root element" but it seemed a bit excessive.